docker/server/entrypoint.sh

```diff
@@ -81,6 +82,7 @@ do
             chown -R "$USER:$GROUP" "$dir"
         fi
     fi
+    cd "$DATA_DIR"
```

Comment:
We are changing to different directories inside a loop, and this is non-obvious.
Our intent is to change to a single directory, once.
Let's put it outside of a loop.

Reply:
Plus add comments.

Reply:
But I'm afraid that the $DATA_DIR directory may not exist before the first loop iteration. 

> changing to different directories inside a loop

It just changes to the same directory $DATA_DIR

Reply:
Some paths relative to $DATA_DIR should be created after `cd` to $DATA_DIR, this is why I put the line just at end of the loop.

Reply:
Let's imaging "${DISKS_PATHS[@]}" and "${DISKS_METADATA_PATHS[@]}" are empty.
Then it will change the working directory to "$FORMAT_SCHEMA_PATH" which looks wrong.

Reply:
No ! It's doing `cd "$DATA_DIR"`, not `cd "$dir"`. The variable $DATA_DIR is a constant in this script, at least not assigned twice.

Reply:
Ok. Does it make sense to do `cd` after the loop?

Reply:
If there're some encrypted disks configuration, "${DISKS_PATHS[@]}" will contain paths of encrypted disk that are [relative](https://github.com/ClickHouse/ClickHouse/blob/587295aca17fc0b4cc852dd429b5aaa9dbc637fb/src/Disks/DiskEncrypted.cpp#L299) to "$DATA_DIR".

Suppose the "$DATA_DIR" directory does not exist yet, the `for` loop creates it at the first iteration, change CWD to "$DATA_DIR", then create subsequent disk paths (including the relative paths of encrypted disks if any).

Reply:
> Ok. Does it make sense to do `cd` after the loop?

Perhaps no. The error #53855 occurs in the loop. It should do `cd` before creating directories of encrypted disks

Reply:
I've taken the `cd $DATA_DIR` out of loop.